### PR TITLE
ATT-48: Complex Obs Encounter Template to handle time zones conversions.

### PR DESCRIPTION
**Pull request by icrc-jfigueiredo**
#### Ticket:
https://issues.openmrs.org/browse/ATT-48
#### What I have done:
* Upgrade UIFR from 3.3.1 to 3.21.0-SNAPSHOT, to access the new `UiUtils` tz API (see [UIFR-219](https://issues.openmrs.org/browse/UIFR-219)).
* Upgrade UICM 1.4 to 2.16.0, to be able to import its Moment.js with locales.
* Using Moment.js with locales instead of the older Moment.js version that was being used (for example to be able to regionalise months names.)

### CODE REVIEW

File: omod/src/main/webapp/fragments/complexObsEncounterTemplate.gsp [hunk 1]
```diff
@@ -71,12 +71,14 @@
 
   <script type="text/template" id="complexObsEncounterTemplate">
   <li>
-    <div class="encounter-date">
+    <div class="encounter-date ${ui.handleTimeZones() ? 'rfc3339-date' : ''}">
```
Comment (mks-d) [inline omod/src/main/webapp/fragments/complexObsEncounterTemplate.gsp]: I think this has become `ui.convertTimezones()` right?

**Comment by dkayiwa** [comment]
@icrc-jfigueiredo did you look into the failing build?

**Comment by icrc-jfigueiredo** [comment]
@dkayiwa i am looking at it right now, it looks like it fails when I increase the UiFramework version

**Comment by dkayiwa** [comment]
@icrc-jfigueiredo did you upgrade the providermanagement module?

**Comment by icrc-jfigueiredo** [comment]
@dkayiwa no, I did not upgrade the providermanagement module

**Comment by dkayiwa** [comment]
@icrc-jfigueiredo upgrading the provider management module is the fix to the failing tests. So if you did not do it, the tests will continue to fail. Just upgrade it in the pom.xml file to any version that you want. That should be all.

**Pull request merged**